include/kodlab_mjbots_sdk/mjbots_simulation_interface.h

```diff
@@ -0,0 +1,359 @@
+    that->mouse_button(window_, button, act, mods);
+  }
+
+  // mouse move callback
+  void mouse_move(GLFWwindow* window_, double xpos, double ypos)
+  {
+      // no buttons down: nothing to do
+      if( !button_left && !button_middle && !button_right )
+          return;
+
+      // compute mouse displacement, save
+      double dx = xpos - lastx;
+      double dy = ypos - lasty;
+      lastx = xpos;
+      lasty = ypos;
+
+      // get current window size
+      int width, height;
+      glfwGetWindowSize(window_, &width, &height);
+
+      // get shift key state
+      bool mod_shift = (glfwGetKey(window_, GLFW_KEY_LEFT_SHIFT)==GLFW_PRESS ||
+                        glfwGetKey(window_, GLFW_KEY_RIGHT_SHIFT)==GLFW_PRESS);
+
+      // determine action based on mouse button
+      mjtMouse action;
+      if( button_right )
+          action = mod_shift ? mjMOUSE_MOVE_H : mjMOUSE_MOVE_V;
+      else if( button_left )
+          action = mod_shift ? mjMOUSE_ROTATE_H : mjMOUSE_ROTATE_V;
+      else
+          action = mjMOUSE_ZOOM;
+
+      // move camera
+      mjv_moveCamera(mj_model, action, dx/height, dy/height, &scn, &cam);
+  }
+  
+  static void mouseMoveCallbackStatic(GLFWwindow* window_, double xpos, double ypos){
+    MujocoAnimator* that = static_cast<MujocoAnimator*>(glfwGetWindowUserPointer(window_));
+    that->mouse_move(window_, xpos, ypos);
+  }
+
+  // scroll callback
+  void scroll(GLFWwindow* window_, double xoffset, double yoffset)
+  {
+      // emulate vertical mouse motion = 5% of window height
+      mjv_moveCamera(mj_model, mjMOUSE_ZOOM, 0, -0.05*yoffset, &scn, &cam);
+  }
+
+  static void scrollCallbackStatic(GLFWwindow* window_, double xoffset, double yoffset){
+    MujocoAnimator* that = static_cast<MujocoAnimator*>(glfwGetWindowUserPointer(window_));
+    that->scroll(window_, xoffset, yoffset);
+  }
+
+};
+}
+
+namespace kodlab::mjbots {
+
+
+
+/*!
+ * @brief Object allowing interaction with the Mjbots Moteus motor controller
+ *        hardware
+ */
+class MjbotsSimulationInterface : public kodlab::RobotInterface {
+ public:
+
+  /*!
+   * @brief constructs an simulation_robot_interface to communicate with MUJOCO simulation
+   * @param joint_list a vector of shared pointers to joints defining the motors in the robot
+   * @param realtime_params the realtime parameters defining cpu and realtime priority
+   * @param imu_mounting_deg Orientation of the imu on the pi3hat. Assumes gravity points in the +z direction
+   * @param imu_rate_hz Frequency of the imu updates from the pi3hat
+   * @param imu_data_ptr Shared pointer to imu_data to use or nullptr if it should make its own
+   * @param imu_world_offset_deg [Optional] IMU orientation offset. Useful for re-orienting gravity, etc.
+   * @param dry_run if true, sends zero-torques to Moteus controllers
+   * @param print_torques if true, prints torque commands
+   * @param send_pd_commands if true, packets to the moteus include pd gains and setpoints
+   */
```

Comment:
Should be Jointbase likely? though not necessarily, just need to make sure the simulation has the ability to face each type of jointsince they have different messages.